### PR TITLE
IntsKey Tests + GetInfo() Clean-up + In-lining

### PR DESCRIPTION
**Pull request by apavlo**
Lots of random stuff in here.

+ New JDBC test that can cause a txn to abort due to conflict. We don't run this automatically. I just wanted to understand how our server-side errors cause exceptions.
+ Inlining two simple methods.
+ Expanded `IndexIntsKeyTest` to look at all combination of integer types
+ New `StringUtil::Format()` helper method that works just like printf.
+ Cleaned up @wangziqi2013's new `CompactIntsKey`
+ Cleaned up `Tuple`

I'm pushing out this PR in memory of @jessesleeping for always being [dangerous](https://youtu.be/pJk0p-98Xzc) with the 9022 bat.

### CODE REVIEW

**Comment by jessesleeping** [comment]
I will order one when I arrive at the Bay Area. It feels so good holding a bat and wandering around in the lab :)

File: src/storage/tuple.cpp [hunk 1]
```diff
@@ -230,14 +230,14 @@ void Tuple::DeserializeFrom(UNUSED_ATTRIBUTE SerializeInput &input,
    * out of Tuple. The memory allocation will be performed when serializing
    * to tuple storage.
    */
-  /*const bool is_inlined = tuple_schema->IsInlined(column_itr);
+  /*const bool is_inlined = tuple_schema_->IsInlined(column_itr);
```
Comment (jessesleeping) [inline src/storage/tuple.cpp]: May be I am wrong but why don't we just remove these dead functions?

Comment (apavlo) [inline src/storage/tuple.cpp]: I saw this but didn't know why it was commented out. So I just left it there for now.

Comment (jessesleeping) [inline src/storage/tuple.cpp]: Hmm... looks like it's some ancient code from last year that used to load tuples from log files or maybe csv/tsv files. I think may be I can revisit it when reviving the logging module.

**Comment by apavlo** [comment]
I have no idea why the build is failing...

**Comment by wangziqi2013** [comment]
@apavlo validator?

**Comment by wangziqi2013** [comment]
Yes it is about the validator

$ ./script/installation/packages.sh

Reading package lists... Done

Building dependency tree       

Reading state information... Done

bison is already the newest version.

flex is already the newest version.

g++ is already the newest version.

g++ set to manually installed.

cmake is already the newest version.

git is already the newest version.

The following extra packages will be installed:

  gdb libboost-atomic1.54-dev libboost-atomic1.54.0 libboost-chrono1.54-dev

  libboost-chrono1.54.0 libboost-date-time1.54-dev libboost-date-time1.54.0

  libboost-filesystem1.54-dev libboost-filesystem1.54.0

  libboost-serialization1.54-dev libboost-serialization1.54.0

  libboost-system1.54-dev libboost-system1.54.0 libboost-thread1.54-dev

  libboost-thread1.54.0 libboost1.54-dev libc6-dbg libevent-core-2.0-5

  libevent-extra-2.0-5 libevent-openssl-2.0-5 libevent-pthreads-2.0-5

  libgd-gd2-perl libgflags2 libjemalloc1 libprotobuf-lite8 libprotobuf8

  libprotoc8

Suggested packages:

  gdb-doc gdbserver libboost-doc libboost1.54-doc libboost-context1.54-dev

  libboost-coroutine.54-dev libboost-exception1.54-dev libboost-graph1.54-dev

  libboost-graph-parallel1.54-dev libboost-iostreams1.54-dev

  libboost-locale1.54-dev libboost-log.54-dev libboost-math1.54-dev

  libboost-mpi1.54-dev libboost-mpi-python1.54-dev

  libboost-program-options1.54-dev libboost-python1.54-dev

  libboost-random1.54-dev libboost-regex1.54-dev libboost-signals1.54-dev

  libboost-test1.54-dev libboost-timer1.54-dev libboost-wave1.54-dev

  libboost1.54-tools-dev libmpfrc++-dev libntl-dev valgrind-dbg kcachegrind

  alleyoop valkyrie

Recommended packages:

  libc-dbg

The following NEW packages will be installed:

  gdb lcov libboost-atomic1.54-dev libboost-atomic1.54.0

  libboost-chrono1.54-dev libboost-chrono1.54.0 libboost-date-time1.54-dev

  libboost-date-time1.54.0 libboost-dev libboost-filesystem-dev

  libboost-filesystem1.54-dev libboost-filesystem1.54.0

  libboost-serialization1.54-dev libboost-serialization1.54.0

  libboost-system1.54-dev libboost-system1.54.0 libboost-thread-dev

  libboost-thread1.54-dev libboost-thread1.54.0 libboost1.54-dev libc6-dbg

  libevent-core-2.0-5 libevent-dev libevent-extra-2.0-5 libevent-openssl-2.0-5

  libevent-pthreads-2.0-5 libgd-gd2-perl libgflags-dev libgflags2

  libjemalloc-dev libjemalloc1 libprotobuf-dev libprotobuf-lite8 libprotobuf8

  libprotoc8 protobuf-compiler valgrind

The following packages will be upgraded:

  postgresql-client

1 upgraded, 37 newly installed, 0 to remove and 64 not upgraded.

Need to get 29.0 MB of archives.

After this operation, 243 MB of additional disk space will be used.

Get:1 http://us-central1.gce.archive.ubuntu.com/ubuntu/ trusty-updates/main libboost-atomic1.54.0 amd64 1.54.0-4ubuntu3.1 [8,076 B]

Get:2 http://us-central1.gce.archive.ubuntu.com/ubuntu/ trusty-updates/main libboost-system1.54.0 amd64 1.54.0-4ubuntu3.1 [10.1 kB]

Get:3 http://us-central1.gce.archive.ubuntu.com/ubuntu/ trusty-updates/main libboost-chrono1.54.0 amd64 1.54.0-4ubuntu3.1 [12.5 kB]

Get:4 http://us-central1.gce.archive.ubuntu.com/ubuntu/ trusty-updates/main libboost-date-time1.54.0 amd64 1.54.0-4ubuntu3.1 [20.8 kB]

Get:5 http://us-central1.gce.archive.ubuntu.com/ubuntu/ trusty-updates/main libboost-filesystem1.54.0 amd64 1.54.0-4ubuntu3.1 [34.2 kB]

Get:6 http://us-central1.gce.archive.ubuntu.com/ubuntu/ trusty-updates/main libboost-serialization1.54.0 amd64 1.54.0-4ubuntu3.1 [126 kB]

Get:7 http://us-central1.gce.archive.ubuntu.com/ubuntu/ trusty-updates/main libboost-thread1.54.0 amd64 1.54.0-4ubuntu3.1 [26.5 kB]

Get:8 http://us-central1.gce.archive.ubuntu.com/ubuntu/ trusty-updates/main libevent-core-2.0-5 amd64 2.0.21-stable-1ubuntu1.14.04.1 [78.3 kB]

Get:9 http://us-central1.gce.archive.ubuntu.com/ubuntu/ trusty-updates/main libevent-extra-2.0-5 amd64 2.0.21-stable-1ubuntu1.14.04.1 [57.8 kB]

Get:10 http://us-central1.gce.archive.ubuntu.com/ubuntu/ trusty-updates/main libevent-openssl-2.0-5 amd64 2.0.21-stable-1ubuntu1.14.04.1 [12.6 kB]

Get:11 http://us-central1.gce.archive.ubuntu.com/ubuntu/ trusty-updates/main libevent-pthreads-2.0-5 amd64 2.0.21-stable-1ubuntu1.14.04.1 [6,126 B]

Get:12 http://us-central1.gce.archive.ubuntu.com/ubuntu/ trusty/main libgflags2 amd64 2.0-1.1ubuntu1 [65.9 kB]

Get:13 http://us-central1.gce.archive.ubuntu.com/ubuntu/ trusty/universe libjemalloc1 amd64 3.5.1-2 [76.8 kB]

Get:14 http://us-central1.gce.archive.ubuntu.com/ubuntu/ trusty-updates/main gdb amd64 7.7.1-0ubuntu5~14.04.2 [2,198 kB]

Get:15 http://us-central1.gce.archive.ubuntu.com/ubuntu/ trusty-updates/main lcov all 1.10-1build1 [89.4 kB]

Get:16 http://us-central1.gce.archive.ubuntu.com/ubuntu/ trusty-updates/main libboost1.54-dev amd64 1.54.0-4ubuntu3.1 [5,682 kB]

Get:17 http://us-central1.gce.archive.ubuntu.com/ubuntu/ trusty-updates/main libboost-atomic1.54-dev amd64 1.54.0-4ubuntu3.1 [4,914 B]

Get:18 http://us-central1.gce.archive.ubuntu.com/ubuntu/ trusty-updates/main libboost-chrono1.54-dev amd64 1.54.0-4ubuntu3.1 [14.2 kB]

Get:19 http://us-central1.gce.archive.ubuntu.com/ubuntu/ trusty-updates/main libboost-serialization1.54-dev amd64 1.54.0-4ubuntu3.1 [151 kB]

Get:20 http://us-central1.gce.archive.ubuntu.com/ubuntu/ trusty-updates/main libboost-date-time1.54-dev amd64 1.54.0-4ubuntu3.1 [25.2 kB]

Get:21 http://us-central1.gce.archive.ubuntu.com/ubuntu/ trusty/main libboost-dev amd64 1.54.0.1ubuntu1 [3,076 B]

Get:22 http://us-central1.gce.archive.ubuntu.com/ubuntu/ trusty-updates/main libboost-system1.54-dev amd64 1.54.0-4ubuntu3.1 [8,428 B]

Get:23 http://us-central1.gce.archive.ubuntu.com/ubuntu/ trusty-updates/main libboost-filesystem1.54-dev amd64 1.54.0-4ubuntu3.1 [41.7 kB]

Get:24 http://us-central1.gce.archive.ubuntu.com/ubuntu/ trusty/main libboost-filesystem-dev amd64 1.54.0.1ubuntu1 [2,854 B]

Get:25 http://apt.postgresql.org/pub/repos/apt/ trusty-pgdg/main postgresql-client all 9.6+178.pgdg14.04+1 [55.3 kB]

Get:26 http://us-central1.gce.archive.ubuntu.com/ubuntu/ trusty-updates/main libboost-thread1.54-dev amd64 1.54.0-4ubuntu3.1 [26.0 kB]

Get:27 http://us-central1.gce.archive.ubuntu.com/ubuntu/ trusty/main libboost-thread-dev amd64 1.54.0.1ubuntu1 [2,848 B]

Get:28 http://us-central1.gce.archive.ubuntu.com/ubuntu/ trusty-updates/main libevent-dev amd64 2.0.21-stable-1ubuntu1.14.04.1 [269 kB]

Get:29 http://us-central1.gce.archive.ubuntu.com/ubuntu/ trusty/main libgflags-dev amd64 2.0-1.1ubuntu1 [72.7 kB]

Get:30 http://us-central1.gce.archive.ubuntu.com/ubuntu/ trusty/main libprotobuf-lite8 amd64 2.5.0-9ubuntu1 [52.7 kB]

Get:31 http://us-central1.gce.archive.ubuntu.com/ubuntu/ trusty/main libprotobuf8 amd64 2.5.0-9ubuntu1 [296 kB]

Get:32 http://us-central1.gce.archive.ubuntu.com/ubuntu/ trusty/main libprotoc8 amd64 2.5.0-9ubuntu1 [235 kB]

Get:33 http://us-central1.gce.archive.ubuntu.com/ubuntu/ trusty-updates/main libc6-dbg amd64 2.19-0ubuntu6.9 [3,450 kB]

Get:34 http://us-central1.gce.archive.ubuntu.com/ubuntu/ trusty-updates/main valgrind amd64 1:3.10.1-1ubuntu3~14.04 [15.0 MB]

Get:35 http://us-central1.gce.archive.ubuntu.com/ubuntu/ trusty/main libgd-gd2-perl amd64 1:2.46-3.1build1 [181 kB]

Get:36 http://us-central1.gce.archive.ubuntu.com/ubuntu/ trusty/universe libjemalloc-dev amd64 3.5.1-2 [164 kB]

Get:37 http://us-central1.gce.archive.ubuntu.com/ubuntu/ trusty/main libprotobuf-dev amd64 2.5.0-9ubuntu1 [421 kB]

Get:38 http://us-central1.gce.archive.ubuntu.com/ubuntu/ trusty/main protobuf-compiler amd64 2.5.0-9ubuntu1 [19.8 kB]

Fetched 29.0 MB in 0s (41.3 MB/s)

Extracting templates from packages: 100%

Selecting previously unselected package libboost-atomic1.54.0:amd64.

(Reading database ... 89283 files and directories currently installed.)

Preparing to unpack .../libboost-atomic1.54.0_1.54.0-4ubuntu3.1_amd64.deb ...

Unpacking libboost-atomic1.54.0:amd64 (1.54.0-4ubuntu3.1) ...

Selecting previously unselected package libboost-system1.54.0:amd64.

Preparing to unpack .../libboost-system1.54.0_1.54.0-4ubuntu3.1_amd64.deb ...

Unpacking libboost-system1.54.0:amd64 (1.54.0-4ubuntu3.1) ...

Selecting previously unselected package libboost-chrono1.54.0:amd64.

Preparing to unpack .../libboost-chrono1.54.0_1.54.0-4ubuntu3.1_amd64.deb ...

Unpacking libboost-chrono1.54.0:amd64 (1.54.0-4ubuntu3.1) ...

Selecting previously unselected package libboost-date-time1.54.0:amd64.

Preparing to unpack .../libboost-date-time1.54.0_1.54.0-4ubuntu3.1_amd64.deb ...

Unpacking libboost-date-time1.54.0:amd64 (1.54.0-4ubuntu3.1) ...

Selecting previously unselected package libboost-filesystem1.54.0:amd64.

Preparing to unpack .../libboost-filesystem1.54.0_1.54.0-4ubuntu3.1_amd64.deb ...

Unpacking libboost-filesystem1.54.0:amd64 (1.54.0-4ubuntu3.1) ...

Selecting previously unselected package libboost-serialization1.54.0:amd64.

Preparing to unpack .../libboost-serialization1.54.0_1.54.0-4ubuntu3.1_amd64.deb ...

Unpacking libboost-serialization1.54.0:amd64 (1.54.0-4ubuntu3.1) ...

Selecting previously unselected package libboost-thread1.54.0:amd64.

Preparing to unpack .../libboost-thread1.54.0_1.54.0-4ubuntu3.1_amd64.deb ...

Unpacking libboost-thread1.54.0:amd64 (1.54.0-4ubuntu3.1) ...

Selecting previously unselected package libevent-core-2.0-5:amd64.

Preparing to unpack .../libevent-core-2.0-5_2.0.21-stable-1ubuntu1.14.04.1_amd64.deb ...

Unpacking libevent-core-2.0-5:amd64 (2.0.21-stable-1ubuntu1.14.04.1) ...

Selecting previously unselected package libevent-extra-2.0-5:amd64.

Preparing to unpack .../libevent-extra-2.0-5_2.0.21-stable-1ubuntu1.14.04.1_amd64.deb ...

Unpacking libevent-extra-2.0-5:amd64 (2.0.21-stable-1ubuntu1.14.04.1) ...

Selecting previously unselected package libevent-openssl-2.0-5:amd64.

Preparing to unpack .../libevent-openssl-2.0-5_2.0.21-stable-1ubuntu1.14.04.1_amd64.deb ...

Unpacking libevent-openssl-2.0-5:amd64 (2.0.21-stable-1ubuntu1.14.04.1) ...

Selecting previously unselected package libevent-pthreads-2.0-5:amd64.

Preparing to unpack .../libevent-pthreads-2.0-5_2.0.21-stable-1ubuntu1.14.04.1_amd64.deb ...

Unpacking libevent-pthreads-2.0-5:amd64 (2.0.21-stable-1ubuntu1.14.04.1) ...

Selecting previously unselected package libgflags2.

Preparing to unpack .../libgflags2_2.0-1.1ubuntu1_amd64.deb ...

Unpacking libgflags2 (2.0-1.1ubuntu1) ...

Selecting previously unselected package libjemalloc1.

Preparing to unpack .../libjemalloc1_3.5.1-2_amd64.deb ...

Unpacking libjemalloc1 (3.5.1-2) ...

Selecting previously unselected package gdb.

Preparing to unpack .../gdb_7.7.1-0ubuntu5~14.04.2_amd64.deb ...

Unpacking gdb (7.7.1-0ubuntu5~14.04.2) ...

Selecting previously unselected package lcov.

Preparing to unpack .../lcov_1.10-1build1_all.deb ...

Unpacking lcov (1.10-1build1) ...

Selecting previously unselected package libboost1.54-dev.

Preparing to unpack .../libboost1.54-dev_1.54.0-4ubuntu3.1_amd64.deb ...

Unpacking libboost1.54-dev (1.54.0-4ubuntu3.1) ...

Selecting previously unselected package libboost-atomic1.54-dev:amd64.

Preparing to unpack .../libboost-atomic1.54-dev_1.54.0-4ubuntu3.1_amd64.deb ...

Unpacking libboost-atomic1.54-dev:amd64 (1.54.0-4ubuntu3.1) ...

Selecting previously unselected package libboost-chrono1.54-dev:amd64.

Preparing to unpack .../libboost-chrono1.54-dev_1.54.0-4ubuntu3.1_amd64.deb ...

Unpacking libboost-chrono1.54-dev:amd64 (1.54.0-4ubuntu3.1) ...

Selecting previously unselected package libboost-serialization1.54-dev:amd64.

Preparing to unpack .../libboost-serialization1.54-dev_1.54.0-4ubuntu3.1_amd64.deb ...

Unpacking libboost-serialization1.54-dev:amd64 (1.54.0-4ubuntu3.1) ...

Selecting previously unselected package libboost-date-time1.54-dev:amd64.

Preparing to unpack .../libboost-date-time1.54-dev_1.54.0-4ubuntu3.1_amd64.deb ...

Unpacking libboost-date-time1.54-dev:amd64 (1.54.0-4ubuntu3.1) ...

Selecting previously unselected package libboost-dev.

Preparing to unpack .../libboost-dev_1.54.0.1ubuntu1_amd64.deb ...

Unpacking libboost-dev (1.54.0.1ubuntu1) ...

Selecting previously unselected package libboost-system1.54-dev:amd64.

Preparing to unpack .../libboost-system1.54-dev_1.54.0-4ubuntu3.1_amd64.deb ...

Unpacking libboost-system1.54-dev:amd64 (1.54.0-4ubuntu3.1) ...

Selecting previously unselected package libboost-filesystem1.54-dev:amd64.

Preparing to unpack .../libboost-filesystem1.54-dev_1.54.0-4ubuntu3.1_amd64.deb ...

Unpacking libboost-filesystem1.54-dev:amd64 (1.54.0-4ubuntu3.1) ...

Selecting previously unselected package libboost-filesystem-dev:amd64.

Preparing to unpack .../libboost-filesystem-dev_1.54.0.1ubuntu1_amd64.deb ...

Unpacking libboost-filesystem-dev:amd64 (1.54.0.1ubuntu1) ...

Selecting previously unselected package libboost-thread1.54-dev:amd64.

Preparing to unpack .../libboost-thread1.54-dev_1.54.0-4ubuntu3.1_amd64.deb ...

Unpacking libboost-thread1.54-dev:amd64 (1.54.0-4ubuntu3.1) ...

Selecting previously unselected package libboost-thread-dev:amd64.

Preparing to unpack .../libboost-thread-dev_1.54.0.1ubuntu1_amd64.deb ...

Unpacking libboost-thread-dev:amd64 (1.54.0.1ubuntu1) ...

Selecting previously unselected package libevent-dev.

Preparing to unpack .../libevent-dev_2.0.21-stable-1ubuntu1.14.04.1_amd64.deb ...

Unpacking libevent-dev (2.0.21-stable-1ubuntu1.14.04.1) ...

Selecting previously unselected package libgflags-dev.

Preparing to unpack .../libgflags-dev_2.0-1.1ubuntu1_amd64.deb ...

Unpacking libgflags-dev (2.0-1.1ubuntu1) ...

Selecting previously unselected package libprotobuf-lite8:amd64.

Preparing to unpack .../libprotobuf-lite8_2.5.0-9ubuntu1_amd64.deb ...

Unpacking libprotobuf-lite8:amd64 (2.5.0-9ubuntu1) ...

Selecting previously unselected package libprotobuf8:amd64.

Preparing to unpack .../libprotobuf8_2.5.0-9ubuntu1_amd64.deb ...

Unpacking libprotobuf8:amd64 (2.5.0-9ubuntu1) ...

Selecting previously unselected package libprotoc8:amd64.

Preparing to unpack .../libprotoc8_2.5.0-9ubuntu1_amd64.deb ...

Unpacking libprotoc8:amd64 (2.5.0-9ubuntu1) ...

Preparing to unpack .../postgresql-client_9.6+178.pgdg14.04+1_all.deb ...

Unpacking postgresql-client (9.6+178.pgdg14.04+1) over (9.3+154ubuntu1) ...

Selecting previously unselected package libc6-dbg:amd64.

Preparing to unpack .../libc6-dbg_2.19-0ubuntu6.9_amd64.deb ...

Unpacking libc6-dbg:amd64 (2.19-0ubuntu6.9) ...

Selecting previously unselected package valgrind.

Preparing to unpack .../valgrind_1%3a3.10.1-1ubuntu3~14.04_amd64.deb ...

Unpacking valgrind (1:3.10.1-1ubuntu3~14.04) ...

Selecting previously unselected package libgd-gd2-perl.

Preparing to unpack .../libgd-gd2-perl_1%3a2.46-3.1build1_amd64.deb ...

Unpacking libgd-gd2-perl (1:2.46-3.1build1) ...

Selecting previously unselected package libjemalloc-dev.

Preparing to unpack .../libjemalloc-dev_3.5.1-2_amd64.deb ...

Unpacking libjemalloc-dev (3.5.1-2) ...

Selecting previously unselected package libprotobuf-dev:amd64.

Preparing to unpack .../libprotobuf-dev_2.5.0-9ubuntu1_amd64.deb ...

Unpacking libprotobuf-dev:amd64 (2.5.0-9ubuntu1) ...

Selecting previously unselected package protobuf-compiler.

Preparing to unpack .../protobuf-compiler_2.5.0-9ubuntu1_amd64.deb ...

Unpacking protobuf-compiler (2.5.0-9ubuntu1) ...

Processing triggers for man-db (2.6.7.1-1ubuntu1) ...

Setting up libboost-atomic1.54.0:amd64 (1.54.0-4ubuntu3.1) ...

Setting up libboost-system1.54.0:amd64 (1.54.0-4ubuntu3.1) ...

Setting up libboost-chrono1.54.0:amd64 (1.54.0-4ubuntu3.1) ...

Setting up libboost-date-time1.54.0:amd64 (1.54.0-4ubuntu3.1) ...

Setting up libboost-filesystem1.54.0:amd64 (1.54.0-4ubuntu3.1) ...

Setting up libboost-serialization1.54.0:amd64 (1.54.0-4ubuntu3.1) ...

Setting up libboost-thread1.54.0:amd64 (1.54.0-4ubuntu3.1) ...

Setting up libevent-core-2.0-5:amd64 (2.0.21-stable-1ubuntu1.14.04.1) ...

Setting up libevent-extra-2.0-5:amd64 (2.0.21-stable-1ubuntu1.14.04.1) ...

Setting up libevent-openssl-2.0-5:amd64 (2.0.21-stable-1ubuntu1.14.04.1) ...

Setting up libevent-pthreads-2.0-5:amd64 (2.0.21-stable-1ubuntu1.14.04.1) ...

Setting up libgflags2 (2.0-1.1ubuntu1) ...

Setting up libjemalloc1 (3.5.1-2) ...

Setting up gdb (7.7.1-0ubuntu5~14.04.2) ...

Setting up lcov (1.10-1build1) ...

Setting up libboost1.54-dev (1.54.0-4ubuntu3.1) ...

Setting up libboost-atomic1.54-dev:amd64 (1.54.0-4ubuntu3.1) ...

Setting up libboost-chrono1.54-dev:amd64 (1.54.0-4ubuntu3.1) ...

Setting up libboost-serialization1.54-dev:amd64 (1.54.0-4ubuntu3.1) ...

Setting up libboost-date-time1.54-dev:amd64 (1.54.0-4ubuntu3.1) ...

Setting up libboost-dev (1.54.0.1ubuntu1) ...

Setting up libboost-system1.54-dev:amd64 (1.54.0-4ubuntu3.1) ...

Setting up libboost-filesystem1.54-dev:amd64 (1.54.0-4ubuntu3.1) ...

Setting up libboost-filesystem-dev:amd64 (1.54.0.1ubuntu1) ...

Setting up libboost-thread1.54-dev:amd64 (1.54.0-4ubuntu3.1) ...

Setting up libboost-thread-dev:amd64 (1.54.0.1ubuntu1) ...

Setting up libevent-dev (2.0.21-stable-1ubuntu1.14.04.1) ...

Setting up libgflags-dev (2.0-1.1ubuntu1) ...

Setting up libprotobuf-lite8:amd64 (2.5.0-9ubuntu1) ...

Setting up libprotobuf8:amd64 (2.5.0-9ubuntu1) ...

Setting up libprotoc8:amd64 (2.5.0-9ubuntu1) ...

Setting up postgresql-client (9.6+178.pgdg14.04+1) ...

Setting up libc6-dbg:amd64 (2.19-0ubuntu6.9) ...

Setting up valgrind (1:3.10.1-1ubuntu3~14.04) ...

Setting up libgd-gd2-perl (1:2.46-3.1build1) ...

Setting up libjemalloc-dev (3.5.1-2) ...

Setting up libprotobuf-dev:amd64 (2.5.0-9ubuntu1) ...

Setting up protobuf-compiler (2.5.0-9ubuntu1) ...

Processing triggers for libc-bin (2.19-0ubuntu6.9) ...

before_script.2

2.02s$ pip install --user cpp-coveralls

Collecting cpp-coveralls

  Downloading cpp-coveralls-0.3.12.tar.gz

Collecting requests (from cpp-coveralls)

  Downloading requests-2.12.4-py2.py3-none-any.whl (576kB)

    100% |████████████████████████████████| 583kB 1.8MB/s 

Installing collected packages: requests, cpp-coveralls

  Running setup.py install for cpp-coveralls ... done

Successfully installed cpp-coveralls requests

You are using pip version 8.1.1, however version 9.0.1 is available.

You should consider upgrading via the 'pip install --upgrade pip' command.

1.14s$ python ./script/validators/validator.py

12-29-2016 03:24:53 [<module>:134] INFO : Scanning : /home/travis/build/cmu-db/peloton/script/validators/../../src

12-29-2016 03:24:54 [<module>:134] INFO : Scanning : /home/travis/build/cmu-db/peloton/script/validators/../../test

12-29-2016 03:24:54 [validate_file:099] INFO : Invalid pattern -- std\:\:cout -- found in : /home/travis/build/cmu-db/peloton/script/validators/../../test/index/index_intskey_test.cpp

12-29-2016 03:24:54 [validate_file:100] INFO : Line ::   // std::cout << "INDEX: " << index->GetInfo() << std::endl;

12-29-2016 03:24:54 [<module>:138] INFO : Validation not successful

**Comment by apavlo** [comment]
@wangziqi2013 Yes you're right. Not sure why the build just doesn't fail right away.

**Comment by wangziqi2013** [comment]
@apavlo It is a feature that exists since long time ago....

**Comment by apavlo** [comment]
I should note that this PR still has the Index::Compare calls. One of the test cases is failing when I take that out. I will fix in the morning.

**Comment by apavlo** [comment]
This doesn't need to be reviewed.

**Pull request merged**